### PR TITLE
Adding spellcheck to CI, and fixing docs along the way

### PR DESCRIPTION
**Pull request by chayim**
**Describe the changes in the pull request**

A clear and concise description of what the PR is solving.

**Which issues this PR fixes**
1. #...
2. MOD...

**Main objects this PR modified**
1. ...
2. ...

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: .github/workflows/spellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Spellcheck
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
```
Comment (DvirDukhan) [inline .github/workflows/spellcheck.yml:6]: No need for version branches?

File: .gitignore [hunk 2]
```diff
@@ -59,3 +59,5 @@ __pycache__/
 
 .vscode
 **/.ipynb_checkpoints
+
+wordlist.dic
```
Comment (DvirDukhan) [inline .gitignore:63]: is this an artifact?

**Review by DvirDukhan — COMMENTED** [review]
thanks @chayim 
1. can you please update PR description
2. should we add "make spellcheck" ?

File: .github/workflows/spellcheck.yml (hunk 1 above)
Comment (chayim) [reply .github/workflows/spellcheck.yml:6]: Added branches.

I don't think we should have a make target - CI will tell us, fairly quickly. If we see it's necessary, let's do it then, but IMHO we shouldn't.

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**